utils/index.js

```diff
@@ -288,9 +297,38 @@ const getPreviousRoute = (req, routes = defaultRoutes) => {
     return oneRouteBack
   }
 
+  if(req.query && req.query.ref) {
+    //I'm assuming here we want to let them go back to yes/no
+    if( req.path.includes('amount') ) {
+      return {path: `${prevRoute().path}?ref=checkAnswers`}
+    }
+    return routes.find(route => route.path === '/checkAnswers')
+  } 
+
   return prevRoute()
 }
 
+const returnToCheckAnswers = (req, res, claimYes = false, routes = defaultRoutes) => {
+
+  const currentRoute = getRouteWithIndexByPath(req.path)
+  const nextRoute = routes[currentRoute.index + 1]
+  const ref = req.query.ref
+  
+  if(
+    ref === 'checkAnswers' && 
+    'editInfo' in nextRoute &&
+    'editInfo' !== 'skip' &&
+    claimYes
+    ) {
+      return res.redirect(`${nextRoute.path}?ref=checkAnswers`)
+    } else if (
+      ref === 'checkAnswers' && 
+      !claimYes 
+    ) {
+      return res.redirect(`/checkAnswers`)
+    }
```

Comment:
This logic is a bit weird because it doesn't cover all cases.

Really we want to say:

- if !checkAnswers (what do we return? error?)
- if editInfo + !skip + claimYes (return path?ref=checkAnswers)
- else (redirect)

Reply:
so we're not really hitting this _unless_ there's a ref of checkAnswers, but I'll add to make that explicit where it could be called (ex in doRedirect and doYesNo)

Reply:
Since we're checking externally now, let's make the checks here less strict. (ie, we don't need to check that ref === `checkAnswers`.)